util/logutil/log.go

```diff
@@ -94,228 +87,41 @@ func NewLogConfig(level, format, slowQueryFile string, fileCfg FileLogConfig, di
-	var b *bytes.Buffer
-	if entry.Buffer != nil {
-		b = entry.Buffer
-	} else {
-		b = &bytes.Buffer{}
-	}
-
-	if !f.DisableTimestamp {
-		fmt.Fprintf(b, "%s ", entry.Time.Format(defaultLogTimeFormat))
-	}
-	if file, ok := entry.Data["file"]; ok {
-		fmt.Fprintf(b, "%s:%v:", file, entry.Data["line"])
-	}
-	fmt.Fprintf(b, " [%s] %s", entry.Level.String(), entry.Message)
-
-	if f.EnableEntryOrder {
-		keys := make([]string, 0, len(entry.Data))
-		for k := range entry.Data {
-			if k != "file" && k != "line" {
-				keys = append(keys, k)
-			}
-		}
-		sort.Strings(keys)
-		for _, k := range keys {
-			fmt.Fprintf(b, " %v=%v", k, entry.Data[k])
-		}
-	} else {
-		for k, v := range entry.Data {
-			if k != "file" && k != "line" {
-				fmt.Fprintf(b, " %v=%v", k, v)
-			}
-		}
-	}
-
-	b.WriteByte('\n')
-
-	return b.Bytes(), nil
-}
-
 const (
 	// SlowLogTimeFormat is the time format for slow log.
 	SlowLogTimeFormat = time.RFC3339Nano
 	// OldSlowLogTimeFormat is the first version of the the time format for slow log, This is use for compatibility.
 	OldSlowLogTimeFormat = "2006-01-02-15:04:05.999999999 -0700"
 )
 
-type slowLogFormatter struct{}
-
-func (f *slowLogFormatter) Format(entry *log.Entry) ([]byte, error) {
-	var b *bytes.Buffer
-	if entry.Buffer != nil {
-		b = entry.Buffer
-	} else {
-		b = &bytes.Buffer{}
-	}
+// SlowQueryLogger is used to log slow query, InitZapLogger will modify it according to config file.
+var SlowQueryLogger = log.L()
 
-	fmt.Fprintf(b, "# Time: %s\n", entry.Time.Format(SlowLogTimeFormat))
-	fmt.Fprintf(b, "%s\n", entry.Message)
-	return b.Bytes(), nil
-}
-
-func stringToLogFormatter(format string, disableTimestamp bool) log.Formatter {
-	switch strings.ToLower(format) {
-	case "text":
-		return &textFormatter{
-			DisableTimestamp: disableTimestamp,
-		}
-	default:
-		return &textFormatter{}
-	}
-}
-
-// initFileLog initializes file based logging options.
-func initFileLog(cfg *zaplog.FileLogConfig, logger *log.Logger) error {
-	if st, err := os.Stat(cfg.Filename); err == nil {
-		if st.IsDir() {
-			return errors.New("can't use directory as log file name")
-		}
-	}
-	if cfg.MaxSize == 0 {
-		cfg.MaxSize = DefaultLogMaxSize
-	}
-
-	// use lumberjack to logrotate
-	output := &lumberjack.Logger{
-		Filename:   cfg.Filename,
-		MaxSize:    cfg.MaxSize,
-		MaxBackups: cfg.MaxBackups,
-		MaxAge:     cfg.MaxDays,
-		LocalTime:  true,
-	}
-
-	if logger == nil {
-		log.SetOutput(output)
-	} else {
-		logger.Out = output
-	}
-	return nil
-}
-
-// SlowQueryLogger is used to log slow query, InitLogger will modify it according to config file.
-var SlowQueryLogger = log.StandardLogger()
-
-// SlowQueryZapLogger is used to log slow query, InitZapLogger will modify it according to config file.
-var SlowQueryZapLogger = zaplog.L()
-
-// InitLogger initializes PD's logger.
+// InitLogger delegates to InitZapLogger. Keeping it here for historical reason.
 func InitLogger(cfg *LogConfig) error {
```

Comment:
PD still depend on this method, right?

Reply:
If so, why not we retain `InitLogger` and drop `InitZapLogger`? An interface is better to be more abstract.